File: 118.pascals-triangle.py
#
# @lc app=leetcode id=118 lang=python
#
# [118] Pascal's Triangle
#

# @lc code=start
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        output = []
        for i in range(numRows):
            output.append([1] * (i + 1))
        for i in range(2, numRows):
            for j in range(1, len(output[i]) - 1):
                output[i][j] = output[i - 1][j - 1] + output[i - 1][j]
        return output
        
# @lc code=end

File: 118.pascals-triangle.java
/*
 * @lc app=leetcode id=118 lang=java
 *
 * [118] Pascal's Triangle
 */

// @lc code=start
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> output = new ArrayList<>();
        for (int i = 0; i<numRows; i++) {
            List<Integer> row = new ArrayList<>();
            for (int j = 0; j<=i; j++) {
                row.add(1);
            }
        output.add(row);
        }

        for (int i = 2; i < numRows; i++) {
            List<Integer> prevRow = output.get(i - 1);
            List<Integer> currRow = output.get(i);
            for (int j = 1; j < currRow.size() - 1; j++) {
                currRow.set(j, prevRow.get(j - 1) + prevRow.get(j));
            }
        }
        return output;
    }
}
// @lc code=end

